apps/modernization-ui/src/apps/deduplication/data-elements/DataElementConfig.tsx

```diff
@@ -0,0 +1,107 @@
+    state: { active: false, oddsRatio: undefined, logOdds: undefined, threshold: undefined },
+    zip: { active: false, oddsRatio: undefined, logOdds: undefined, threshold: undefined },
+    county: { active: false, oddsRatio: undefined, logOdds: undefined, threshold: undefined },
+    telecom: { active: false, oddsRatio: undefined, logOdds: undefined, threshold: undefined },
+    telephone: { active: false, oddsRatio: undefined, logOdds: undefined, threshold: undefined },
+    email: { active: false, oddsRatio: undefined, logOdds: undefined, threshold: undefined },
+    // Identification Details
+    accountNumber: { active: false, oddsRatio: undefined, logOdds: undefined, threshold: undefined },
+    driversLicenseNumber: { active: false, oddsRatio: undefined, logOdds: undefined, threshold: undefined },
+    medicaidNumber: { active: false, oddsRatio: undefined, logOdds: undefined, threshold: undefined },
+    medicalRecordNumber: { active: false, oddsRatio: undefined, logOdds: undefined, threshold: undefined },
+    medicareNumber: { active: false, oddsRatio: undefined, logOdds: undefined, threshold: undefined },
+    nationalUniqueIdentifier: { active: false, oddsRatio: undefined, logOdds: undefined, threshold: undefined },
+    patientExternalIdentifier: { active: false, oddsRatio: undefined, logOdds: undefined, threshold: undefined },
+    patientInternalIdentifier: { active: false, oddsRatio: undefined, logOdds: undefined, threshold: undefined },
+    personNumber: { active: false, oddsRatio: undefined, logOdds: undefined, threshold: undefined },
+    socialSecurity: { active: false, oddsRatio: undefined, logOdds: undefined, threshold: undefined },
+    visaPassport: { active: false, oddsRatio: undefined, logOdds: undefined, threshold: undefined },
+    wicIdentifier: { active: false, oddsRatio: undefined, logOdds: undefined, threshold: undefined }
+};
+
+const DataElementConfigContent = () => {
+    const { showSuccess, showError } = useAlert();
+    const { configuration, save, error } = useDataElements();
+    const form = useForm<DataElements>({ mode: 'onBlur', defaultValues: initial });
+    const nav = useNavigate();
+
+    useEffect(() => {
+        form.reset(configuration, { keepDefaultValues: false, keepDirty: false });
+    }, [configuration]);
+
+    useEffect(() => {
+        if (error) {
+            showError({ message: error });
+        }
+    }, [error]);
+
+    const handleCancel = () => {
+        nav({ pathname: '/deduplication/configuration' });
+    };
+    const handleSubmit = () => {
+        save(form.getValues(), () =>
+            showSuccess({ message: 'You have successfully updated the data elements configuration.' })
+        );
+    };
+
+    const allValues = form.watch(); // Use watch to get live updates
+
+    const hasInvalidValues = Object.values(allValues).some(
+        (element) => element.active && (!element.oddsRatio || !element.threshold)
+    );
```

Comment:
Suggestion: If you have each field setup with a `Controller` that has the proper validation rules assigned, you should be able to use the form's built in validation.

```typescript
const { isValid } = useFormState(form);
```